src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,224 @@
+  def actorRefFactory: ActorContext = context
   override def receive: Receive = {
-    case ExportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) => exportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) pipeTo sender
+    case ExportEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) => streamEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) pipeTo sender
   }
 }
 
 trait ExportEntitiesByType extends FireCloudRequestBuilding {
   val rawlsDAO: RawlsDAO
+  val googleDAO: GoogleServicesDAO
   implicit val userInfo: UserInfo
   implicit protected val executionContext: ExecutionContext
+  implicit def actorRefFactory: ActorRefFactory
 
-  def exportEntities(workspaceNamespace: String, workspaceName: String, filename: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[PerRequestMessage] = {
-    rawlsDAO.fetchAllEntitiesOfType(workspaceNamespace, workspaceName, entityType) map { entities =>
-      ModelSchema.getCollectionMemberType(entityType) match {
-        case Success(Some(collectionType)) =>
-          val collectionMemberType = ModelSchema.getPlural(collectionType)
-          val entityData = TSVFormatter.makeEntityTsvString(entities, entityType, attributeNames)
-          val membershipData = TSVFormatter.makeMembershipTsvString(entities, entityType, collectionType, collectionMemberType.get)
-          val zipBytes: Array[Byte] = getZipBytes(entityType, membershipData, entityData)
-          val zippedFileName = entityType + ".zip"
-          RequestCompleteWithHeaders(
-            HttpEntity(ContentTypes.`application/octet-stream`, zipBytes),
-            HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> zippedFileName)))
-        case _ =>
-          val data = TSVFormatter.makeEntityTsvString(entities, entityType, attributeNames)
-          RequestCompleteWithHeaders(
-            (OK, data),
-            HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> filename)),
-            HttpHeaders.`Content-Type`(`text/plain`))
+  implicit val timeout = Timeout(1 minute)
+
+  lazy val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  /**
+    * Two basic code paths
+    *
+    * For Collection types, write the content to temp files, zip and return.
+    *
+    * For Singular types, pipe the content from `Source` -> `Flow` -> `Sink`
+    *   Source generates the entity queries
+    *   Flow executes the queries and sends formatted content to chunked response handler
+    *   Sink finishes the execution pipeline
+    *
+    * Handle exceptions directly by completing the request.
+    */
+  def streamEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[Any] = {
```

Comment:
I see -- no one is even using the return value of this method because you're completing the RequestContext directly. It looks like the error case returns a `Future[Unit]` (because `ctx.complete` returns Unit), and the success case returns a `Future[Done]`. Maybe you could map the `Done` to a `Unit` and have the whole method return a `Future[Unit]`.